src/main/java/net/dv8tion/jda/internal/entities/UserById.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2015-2020 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.entities;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.PrivateChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.Contract;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.EnumSet;
+import java.util.List;
+
+public class UserById implements User
+{
+    protected final long id;
+
+    public UserById(long id)
+    {
+        this.id = id;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return this.id;
+    }
+
+    @Nonnull
+    @Override
+    public String getAsMention()
+    {
+        return "<@" + getId() + ">";
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Long.hashCode(id);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == this)
+            return true;
+        if (!(obj instanceof User))
+            return false;
+        return ((User) obj).getIdLong() == this.id;
+    }
+
+    @Override
+    public String toString()
+    {
+        return "U:(" + getId() + ')';
+    }
+
+    @Contract("->fail")
+    private void unsupported()
+    {
+        throw new UnsupportedOperationException("This User instance only wraps an ID. Other operations are unsupported");
+    }
+
+    @Nonnull
+    @Override
+    public String getName()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getDiscriminator()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public String getAvatarId()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getDefaultAvatarId()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getAsTag()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Override
+    public boolean hasPrivateChannel()
+    {
+        unsupported();
+        return false;
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<PrivateChannel> openPrivateChannel()
+    {
+        unsupported();
```

Comment:
I feel like, that this method should be available, because this request only requires the already provided ID (and a JDA instance).
I am aware that this breaks the pattern of JDA, that JDA you would need to manually pass a JDA reference for being able to execute this request.

I don't see why an extra request for retrieving the user by id should be required if you have user / member cache disabled. I know that this is a temporarily created user object that gets collected by the gc fairly quickly, but it still feels unnecessary, if it's possible to achieve this without an extra footprint, if this low memory profile wrapper exists in the future.

Reply:
You can just use JDA#openPrivateChannelById instead

Reply:
Let's pretend this didn't happen.